lib/perf_hooks.js

```diff
@@ -396,11 +396,12 @@ class Performance {
   measure(name, startMark, endMark) {
     name = `${name}`;
     endMark = `${endMark}`;
-    startMark = startMark !== undefined ? `${startMark}` : '';
     const marks = this[kIndex][kMarks];
-    if (!marks.has(endMark) && !(endMark in nodeTiming)) {
+    if (arguments.length === 3 &&
+        !marks.has(endMark) && !(endMark in nodeTiming)) {
       throw new ERR_INVALID_PERFORMANCE_MARK(endMark);
     }
+    startMark = startMark !== undefined ? `${startMark}` : '';
     _measure(name, startMark, endMark);
```

Comment:
This passes the string `'undefined'` to `_measure` if `endMark` was not passed – that’s not what you want, is it? What if there is a mark in the table with that name?

Reply:
not exactly, It passes '' (empty string) to `_measure`, and previous code already use `timeOrigin` when not match the mark, so I thought to keep origin behavior is ok

https://github.com/nodejs/node/blob/821c6860df83cc4a16ee5b85e29a0165ea43d224/src/node_perf.cc#L179

Reply:
@Himself65 If I checkout the code in this PR and add

```diff
--- a/src/node_perf.cc
+++ b/src/node_perf.cc
@@ -174,4 +174,5 @@ void Measure(const FunctionCallbackInfo<Value>& args) {
   Utf8Value startMark(env->isolate(), args[1]);
   Utf8Value endMark(env->isolate(), args[2]);
+  fprintf(stderr, "endMark = '%s'\n", *endMark);
 
   AliasedFloat64Array& milestones = env->performance_state()->milestones;
```

then it prints

```
endMark = 'undefined'
```

for `perf_hooks.performance.measure("A")`, as predicted.

Reply:
Oh, no. I thought you said `startMark`. Working on this

Reply:
fixed now.